plotly/tests/test_optional/optional_utils.py

```diff
@@ -29,6 +29,25 @@ def _format_path(self, path):
         str_path = [repr(p) for p in path]
         return '[' + ']['.join(sp for sp in str_path) + ']'
 
+    def assert_fig_equal(self, d1, d2, msg=None, ignore=['uid']):
+        """
+        Helper function for asser_dict_equal
+
+        By defualt removes uid from d1 and/or d2 if present
+        then calls assert_dict_equal.
+
+        :param (list|tuple) ignore: sequence of key names as
+            strings that are removed from both d1 and d2 if
+            they exist
+        """
+        for key in ignore:
```

Comment:
We might want to make a copy (`import copy; copy.deepcopy` i think) before deleting, that way there are no surprises from the callers code. Otherwise, the user has something like:
```
my_scatter = {'x': 1, 'uid': 5}
assert_fig_equal(my_scatter, ...)
print(my_scatter) # is now just {'x': 1}
```